app/models/city.rb

```diff
@@ -22,6 +25,6 @@ def slug
   end
 
   def top_contributors
-    [10, (size * 0.03).ceil].max
+    [10, (batches.sum(&:size) * 0.03).ceil].max
```

Comment:
Si on se repose plus sur un attribut hardcodé, on voudra peut-être mémoiser ça.
```suggestion
    @top_contributors ||= [10, (batches.sum(&:size) * 0.03).ceil].max
```
Ça fera pas _grand_ chose mais au moins appeler 3 fois la méthode refera pas le calcul trois fois (qui est plus coûteux que juste une multiplication, vu qu'on itère sur tous les batchs).

On peut aussi faire la somme en sql direct en utilisant `sum(<colonne>)` au lieu de `sum(&block)`:
```suggestion
    @top_contributors ||= [10, (batches.sum(:size) * 0.03).ceil].max
```

Reply:
En vrai la question se pose de garder la colonne juste pour faciliter ce genre de requête / calcul 🤔 
(Et qu'elle soit remplie via le code de création des batchs, ex: dans un `after_create`/`before_destroy`)

Reply:
Si  tu store dans une variable et qu'ensuite tu fais ||= la variable est lock nan ?
L'année prochaine quand il y aura plus de batch, dont d'alumnis, le sum(:size) aura changé donc il faudra réavaluer le @top_contributors.

Dans le code que tu propose j'ai l'impression que le nombre de contributors s'adaptera pas automatiquement a l'ajout de nouveau batch plus tard mais je comprend peut etre pas comment ||= fonctionne